sqlglot/optimizer/scope.py

```diff
@@ -122,7 +122,11 @@ def _collect(self):
                 self._udtfs.append(node)
             elif isinstance(node, exp.CTE):
                 self._ctes.append(node)
-            elif isinstance(node, exp.Subquery) and isinstance(parent, (exp.From, exp.Join)):
+            elif (
+                isinstance(node, exp.Subquery)
+                and isinstance(parent, (exp.From, exp.Join))
+                and _is_subquery_scope(node)
+            ):
```

Comment:
Since nested joins such as `(t1 JOIN t2)` don't introduce a new scope, `walk` will visit them too, so we want to make sure they won't be treated as derived tables.